lib/storage/client.js

```diff
@@ -1182,6 +1183,64 @@ class Client extends CryptoConsumer {
     })
     return listGroups
   }
-}
 
+  /**
+   * addGroupMembers adds members to the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   * @param {GroupMember[]} groupMembers wraps together the ClientID, Capability and Membership Key for each client being added to the group
+   *
+   *
+   * @returns {GroupMember[]} A response from TozStore; the group members who have been added.
+   *
+   */
+  async addGroupMembers(groupId, groupMembers) {
+    if (!(groupMembers instanceof GroupMember)) {
+      throw new Error('Need an array of group members')
+    }
+    // Take the Given Paramaters and create the object needed for endpoint
+    let addGroupMembersRequest = {
+      group_id: groupId,
+      group_members: groupMembers,
+    }
+    const addGroupMembersRequestKeys = Object.keys(addGroupMembersRequest)
+    for (const key of addGroupMembersRequestKeys) {
+      if (
+        addGroupMembersRequest[key] === null ||
+        addGroupMembersRequest[key] === []
+      ) {
+        delete addGroupMembersRequest[key]
+      }
+    }
+    // Call the API endpoint to add group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(addGroupMembersRequest),
+      }
+    )
+    // Await the Response from the API with the Added Group member Response
+    let groupMemberJson = await validateResponseAsJSON(response)
+    let listGroupMembers = []
+    groupMemberJson.forEach(groupMember => {
+      listGroupMembers.push(GroupMember.decode(groupMember))
+    })
+    switch (response.status) {
```

Comment:
Take a look at what `validateResponseAsJSON` does, I think that because it catches any non 2xx response and throws an exception that you can and should remove the switch statement as you're only going to make it past line 1227 if the response is a 2xx level. (Which obviously I'm opposed to because it means that we've given up control to a function we call unless we want to wrap and catch the exception ourself, but as with everything, tradeoffs and time make pragmatists of us all, so it is what it is and we can work around it)